packages/jest-editor-support/src/SnapshotSupport.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const babylon = require('babylon');
+const fs = require('fs');
+const path = require('path');
+const walker = require('./walker');
+
+type Node = any;
+
+type SnapshotMetadata = {
+  exists: true | false,
+  exportName: string,
+  node: Node,
+  content?: string,
+};
+
+type SnapshotSupportType = {
+  getMetadataFromTestFile: (filePath: string) => Array<SnapshotMetadata>,
+  parser?: (content: string, options: any) => Node,
+  plugins?: Array<string>,
+};
+
+const buildExportName: (
+  toMatchSnapshot: Node,
+  parents: Array<Node>
+) => string = (toMatchSnapshot, parents) => {
+  const state = {
+    found: [],
+  };
+
+  walker(
+    parents[parents.length - 1],
+    {
+      Identifier(node, state, parents) {
+        if (node.name === 'toMatchSnapshot') {
+          state.found.push({node, parents});
+        }
+      },
+    },
+    state
+  );
+
+  const position = state.found.findIndex(
+    node => node.node === toMatchSnapshot.node
+  ) + 1;
+
+  const fullName = parents.map(parent => parent.arguments[0].value).join(' ');
+
+  let describeLess = '';
+  if (parents[0].callee.name !== 'describe') {
+    // if `it` or `test` exists without a surrounding `describe`
+    // then `test ` is prepended to the snapshot fullName
+    describeLess = 'test ';
+  }
+
+  return `${describeLess}${fullName} ${position}`;
+};
+
+const snapshotPathFromFilePath = filePath => (
+  path.join(
+    path.dirname(filePath),
+    '__snapshots__',
+    path.basename(filePath) + '.snap'
+  )
+);
+
+const SnapshotSupport: SnapshotSupportType = {
+  getMetadataFromTestFile(filePath: string): Array<SnapshotMetadata> {
```

Comment:
I'd recommend calling this something like:

`Snapshot.getMetadata(…)`.

I don't think either the "fromTestFile" or "Support" need to be there. The editor support package already signals it is a "support package" and we only have one getter for metadata. If we ever add different files that may have snapshots (like regular files), then it makes sense to split it up into `getMetadataFromTestFile` and `getMetadataFromMockFile` for example.